scripts/xtensa-build-all.sh

```diff
@@ -113,171 +90,148 @@ set -e
 OLDPATH=$PATH
 WORKDIR="$pwd"
 
-# build platform
-for j in ${PLATFORMS[@]}
+# build platforms
+for platform in ${PLATFORMS[@]}
 do
 	HAVE_ROM='no'
 	DEFCONFIG_PATCH=''
-	if [ $j == "byt" ]
-	then
-		PLATFORM="baytrail"
-		ARCH="xtensa"
-		XTENSA_CORE="Intel_HiFiEP"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
-		HOST="xtensa-byt-elf"
-		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
-	fi
-	if [ $j == "cht" ]
-	then
-		PLATFORM="cherrytrail"
-		ARCH="xtensa"
-		XTENSA_CORE="CHT_audio_hifiep"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
-		HOST="xtensa-byt-elf"
-		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
-	fi
-	if [ $j == "bdw" ]
-	then
-		PLATFORM="broadwell"
-		ARCH="xtensa"
-		XTENSA_CORE="LX4_langwell_audio_17_8"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
-		HOST="xtensa-hsw-elf"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-	fi
-	if [ $j == "hsw" ]
-	then
-		PLATFORM="haswell"
-		ARCH="xtensa"
-		XTENSA_CORE="LX4_langwell_audio_17_8"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
-		HOST="xtensa-hsw-elf"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-	fi
-	if [ $j == "apl" ]
-	then
-		PLATFORM="apollolake"
-		ARCH="xtensa-smp"
-		XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
 
-		# test APL compiler aliases and ignore set -e here
-		type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
-		if [ $? == 0 ]
-		then
-			HOST="xtensa-bxt-elf"
-		else
-			HOST="xtensa-apl-elf"
-		fi
-
-		ROOT="$pwd/../xtensa-root/$HOST"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-		HAVE_ROM='yes'
-	fi
-	if [ $j == "skl" ]
-	then
-		PLATFORM="skylake"
-		ARCH="xtensa"
-		XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
-
-		# test APL compiler aliases and ignore set -e here
-		type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
-		if [ $? == 0 ]
-		then
-			HOST="xtensa-bxt-elf"
-		else
-			HOST="xtensa-apl-elf"
-		fi
+	case $platform in
+		byt)
+			PLATFORM="baytrail"
+			ARCH="xtensa"
+			XTENSA_CORE="Intel_HiFiEP"
+			HOST="xtensa-byt-elf"
+			XTENSA_TOOLS_VERSION="RD-2012.5-linux"
+			;;
+		cht)
+			PLATFORM="cherrytrail"
+			ARCH="xtensa"
+			XTENSA_CORE="CHT_audio_hifiep"
+			HOST="xtensa-byt-elf"
+			XTENSA_TOOLS_VERSION="RD-2012.5-linux"
+			;;
+		bdw)
+			PLATFORM="broadwell"
+			ARCH="xtensa"
+			XTENSA_CORE="LX4_langwell_audio_17_8"
+			HOST="xtensa-hsw-elf"
+			XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+			;;
+		hsw)
+			PLATFORM="haswell"
+			ARCH="xtensa"
+			XTENSA_CORE="LX4_langwell_audio_17_8"
+			HOST="xtensa-hsw-elf"
+			XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+			;;
+		apl)
+			PLATFORM="apollolake"
+			ARCH="xtensa-smp"
+			XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
+
+			# test APL compiler aliases and ignore set -e here
+			type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
+			if [ $? == 0 ]
```

Comment:
```
if type -p xtensa-bxt-elf-gcc ; then
  ...
```
shellcheck should flag this.

The redirections don't seem useful to me.

Same below.

Reply:
If we don't redirect we end up with the path of `xtensa-bxt-elf-gcc` somewhere in the output?

Reply:
Yes and I think it's useful information, think someone testing multiple toolchains version or a developer trying to make sense of an unknown docker image in CI. For someone who has only one toolchain this is just one extra line in an otherwise quite verbose log already. Either way, no big deal.

I've almost never encountered a situation where `/dev/null` is actually useful in a build script, I think it's just a really bad habit IMHO. I've seen `/dev/null` used even when the command has a `--quiet` option... Some tools (not `type`) don't use stderr, they print errors on stdout too then good luck when something goes wrong. If something is really too noisy and can't or shouldn't be turned off it should be sent to a log file or to a lower log level.

To give a bit more perspective: some CI engineers run almost all their CI scripts with `set -x`. They've lost too much time in the past trying to figure out some subtle change/difference in some random OS image.